Add feature specification template to design repository (#55)
radius-project/design-notes: template/YYYY-MM-feature-spec-template.md

# Topic: TODO

* **Status**: Pending/Approved
* **Author**: Your name (@YourGitHubUserName)

## Topic Summary
<!-- A paragraph or two to summarize the topic area. Just define it in summary form so we all know what it is. -->

### Top level goals
<!-- At the most basic level, what are we trying to accomplish? -->

### Non-goals (out of scope)
<!-- What are we explicitly not trying to accomplish? -->

> ## Customer profile and challenges
> <!-- Define the primary customer and the key problem / pain point we intend to address for that customer. If there are multiple customers or primary and secondary customers, call them out.   -->

Reviewer: ```suggestion
## User profile and challenges
<!-- Define the primary user and the key problem / pain point we intend to address for that user. If there are multiple users or primary and secondary users, call them out.   -->
```

> ### Customer persona(s)
> <!-- Who is the target customer? Include size/org-structure/decision makers where applicable. -->

Reviewer: ```suggestion
### User persona(s)
<!-- Who is the target user? Include size/org-structure/decision makers where applicable. -->
```

> ### Challenge(s) faced by the customer
> <!-- What challenges do the customer face? Why are they experiencing pain and why do current offerings not meet their need? -->

Reviewer: ```suggestion
### Challenge(s) faced by the user
<!-- What challenges do the user face? Why are they experiencing pain and why do current offerings not meet their need? -->
```

> ### Positive customer outcome
> <!-- What is the positive outcome for the customer if we deliver this, i.e. what is the value proposition for the customer? Remember, this is customer-centric. -->

Reviewer: ```suggestion
### Positive user outcome
<!-- What is the positive outcome for the user if we deliver this, i.e. what is the value proposition for the user? Remember, this is user-centric. -->
```

## Key scenarios
<!-- List ~3-7 high level scenarios to clarify the value and point to how we will decompose this big area into component capabilities. We may ultimately have more than one level of scenario. -->

### Scenario 1: TODO
<!-- One or two sentence summary -->

### Scenario 2: TODO
<!-- One or two sentence summary -->

### Scenario 3: TODO
<!-- One or two sentence summary -->

## Key dependencies and risks
<!-- What dependencies must we take in order to enable this scenario? -->
<!-- What other risks are you aware of that need to be mitigated. If you have a mitigation in mind, summarize here. -->
<!-- **Dependency Name** – summary of dependency.  Issues/concerns/risks with this dependency -->
<!-- **Risk Name** – summary of risk.  Mitigation plan if known. If it is not yet known, no problem. -->

> ## Key assumptions to test and questions to answer
> <!-- If you are making assumptions that, if incorrect, would cause us to significantly alter our approach to this scenario, make them explicit here.  Also call out how / when you plan to validate key assumptions. -->
> <!-- What big questions must we answer in order to clarify our plan for this scenario.  When and how do you plan to answer those questions (prototype feature x, customer research, competitive research, etc) -->

Reviewer: ```suggestion
## Key assumptions to test and questions to answer
<!-- If you are making assumptions that, if incorrect, would cause us to significantly alter our approach to this scenario, make them explicit here.  Also call out how / when you plan to validate key assumptions. -->
<!-- What big questions must we answer in order to clarify our plan for this scenario.  When and how do you plan to answer those questions (prototype feature x, user research, competitive research, etc.) -->
```

## Current state
<!-- If we already have some ongoing investment in this area, summarize the current state and point to any relevant documents. -->

> ## Details of customer problem
> <!-- <Write this in first person. You basically want to summarize what “I” as a customer am trying to accomplish, why the current experience is a problem and the impact it has on me, my team, my work and or biz, etc…. i.e. “When I try to do x aspect of cloud native app development, I have the following challenges / issues….<details>. Those issues result in <negative impact those challenges / issues have on your work and or business.> -->
> 
> ## Desired customer experience outcome
> <!-- <Write this as an “I statement” that expresses the new capability from customer perspective … i.e. After this scenario is implemented “I can do, x, y, z, steps in cloud native app developer and seamlessly blah blah blah …. As a result <summarize positive impact on your work / business>  -->
> 
> ### Detailed Customer Experience
>  <!-- <List of steps the customer goes through from the start to the end of the scenario to provide more detailed view of exactly what the customer is able to do given the new capabilities>  -->
> <!-- Step 1
> Step 2
> … -->

Reviewer: ```suggestion
## Details of user problem
<!-- <Write this in first person. You basically want to summarize what “I” as a user am trying to accomplish, why the current experience is a problem and the impact it has on me, my team, my work and or biz, etc…. i.e. “When I try to do x aspect of cloud native app development, I have the following challenges / issues….<details>. Those issues result in <negative impact those challenges / issues have on my work and or business.> -->

## Desired user experience outcome
<!-- <Write this as an “I statement” that expresses the new capability from user perspective … i.e. After this scenario is implemented “I can do, x, y, z, steps in cloud native app developer and seamlessly etc... As a result <summarize positive impact on my work / business>  -->

### Detailed user experience
 <!-- <List of steps the user goes through from the start to the end of the scenario to provide more detailed view of exactly what the user is able to do given the new capabilities>  -->
<!-- Step 1
Step 2
… -->
```

## Key investments

> <!-- List the features required to enable this scenario. -->

Reviewer: ```suggestion
<!-- List the features required to enable this scenario(s). -->
```
